packages/server/src/api/controllers/application.ts

```diff
@@ -172,16 +171,16 @@ export const fetch = async (ctx: any) => {
   const all = ctx.query && ctx.query.status === AppStatus.ALL
   const apps = await getAllApps({ dev, all })
 
+  const appIds = apps
+    .filter((app: any) => app.status === "development")
+    .map((app: any) => app.appId)
   // get the locks for all the dev apps
   if (dev || all) {
-    const locks = await getAllLocks()
+    const locks = await getLocksById(appIds)
     for (let app of apps) {
-      if (app.status !== "development") {
-        continue
-      }
-      const lock = locks.find((lock: any) => lock.appId === app.appId)
+      const lock = locks[app.appId]
       if (lock) {
-        app.lockedBy = lock.user
+        app.lockedBy = lock
```

Comment:
Has the structure of what's in the lock changed?

Reply:
Yeah - previously the `getAllLocks` return an array, of just all the locks by `{ key: appId, user }` - instead now the Redis system returns an object map of appId to user - may as well do it is way for bulk get of keys rather than array.